apps/ui/src/networks/offchain/api/index.ts

```diff
@@ -329,18 +331,42 @@ export function createApi(uri: string, networkId: NetworkID): NetworkApi {
       { limit, skip = 0 }: PaginationOpts,
       filter?: SpacesFilter
     ): Promise<Space[]> => {
+      let _filters = clone(filter || {});
+
+      if (!_filters?.id_in) {
+        const { data } = await apollo.query({
+          query: RANKING_QUERY,
+          variables: {
+            first: limit,
+            skip,
+            where: {
+              ..._filters
+            }
+          }
+        });
+
+        _filters = { id_in: data.ranking.items.map(space => space.id) };
+        skip = 0;
+      }
```

Comment:
I'm not 100% familiar with what we are changing there, but do we need to make two requests?

I was thinking we would look at `_filters?.id_in` and:
1. if `_filters?.id_in` is there we use non-ranking query (`SPACES_QUERY`) (because ranking query doesn't work with `id_in` filter AFAIK)  and return whatever we got right away.
2. if `_filters?.id_in` is not used we use use ranking query and return what we got (so same behaviour as before).

Is this something we can do? Or it's not possible for some reason?